overtime/algorithms/edgeDeletion.py

```diff
@@ -0,0 +1,285 @@
+    end = timespan[-1]  # end time.
+
+    # initialize the foremost tree object.
+    tree = ot.ForemostTree(graph.label, root, start)
+
+    # add each node in the graph to the foremost tree.
+    # nodes in the foremost tree are of type ForemostNode and include a time property,
+    # which initializes at inf.
+    for node in graph.nodes.set:
+        tree.nodes.add(node.label)
+
+    # calculate the number of vertices in the tree
+    accumulator = 0
+
+    # foremost path algorithm:
+    # for every edge in the graph edges set (ordered by edge duration start times).
+    for edge in graph.edges.set:
+        departure = tree.nodes.get(edge.source.label)  # departure node of edge
+        destination = tree.nodes.get(edge.sink.label)  # destination node of edge
+
+        # if edge's duration end is less than or equal to the end time of the graph's timespan
+        # and edge's duration start is greater than or equal to the departure node's foremost time.
+        # else if edge's duration start is greater than or equal to the end time of the graph's timespan.
+        if edge.end <= end and edge.start >= departure.time:
+            # if edge's time duration end is less than the destination node's foremost time.
+            if edge.end < destination.time:
+                # add this edge to the foremost tree.
+                temp = tree.edges.add(edge.source.label, edge.sink.label, tree.nodes, edge.start, edge.end)
+                # update the destination node's foremost time.
+                destination.time = edge.end
+                # update the destination node's data.
+                destination.data['foremost_time'] = edge.end
+
+                # update accumulator
+                accumulator += 1
+                if accumulator == h:
+                    break
+
+        elif edge.start >= end:
+            # stop the algorithm.
+            break
+
+    # return the resulting foremost tree.
+    return tree
+
+
+def h_approximation(graph, h):
+    """
+        A h-approximation algorithm to return a set of edges E_ such that (G,λ)\E_ has temporal reachability at most h.
+
+        Parameter(s):
+        -------------
+        graph : TemporalDiGraph
+                A directed, temporal graph.
+        h : int
+            the maximum permitted temporal reachability.
+
+        Returns:
+        --------
+        E_ : list
+            a set of edges such that (G,λ)/E_ has temporal reachability at most h.
+
+    """
+
+    E_ = []
+
+    # find the node whose temporal reachability is more than h
+    while (True):
+        root = ''
+        for node in graph.nodes.set:
+            reachability = ot.calculate_reachability(graph, node.label)
+            if reachability > h:
+                root = node.label
+                break
+
+        # check if the specified root actually exists in the graph.
+        if root == '':
+            print('there is no node whose reachability is more than {}'.format(h))

```

Comment:
Change the text to
"The temporal reachability of the input temporal graph is at most h"